extensions/amp-story/1.0/page-advancement.js

```diff
@@ -289,32 +304,78 @@ class ManualAdvancement extends AdvancementConfig {
   }
 
   /** @override */
-  start() {
-    super.start();
-    this.element_.addEventListener('click', this.clickListener_, true);
+  getProgress() {
+    return 1.0;
   }
 
-  /** @override */
-  stop() {
-    super.stop();
-    this.element_.removeEventListener('click', this.clickListener_, true);
+  /**
+   * Binds the event listeners.
+   * @private
+   */
+  startListening_() {
+    if (isExperimentOn(this.win_, 'amp-story-hold-to-pause')) {
+      this.element_
+          .addEventListener('touchstart', this.onTouchstart_.bind(this), true);
+      this.element_
+          .addEventListener('touchend', this.onTouchend_.bind(this), true);
+    }
+
+    this.element_
+        .addEventListener(
+            'click', this.maybePerformNavigation_.bind(this), true);
   }
 
-  /** @override */
-  getProgress() {
-    return 1.0;
+  /**
+   * TouchEvent touchstart events handler.
+   * @param {!Event} event
+   * @private
+   */
+  onTouchstart_(event) {
+    // Don't start the paused state if the target is not a descendant of an
+    // amp-story-page. Also ignores any subsequent touchstart that would happen
+    // before touchend was fired, since it'd reset the touchstartTimestamp (ie:
+    // user touches the screen with a second finger).
+    if (this.touchstartTimestamp_ || !this.isAmpStoryPageDescendant_(event)) {
+      return;
+    }
+
+    this.touchstartTimestamp_ = Date.now();
+    this.storeService_.dispatch(Action.TOGGLE_PAUSED, true);
```

Comment:
I think we should add a new state+action (maybe in a separate PR) to represent the "Paused UI", which hides the system layer after N ms (where N ≈ 1000).

Reply:
I have this PR waiting (because no diffbase on github :( ) where I add a `TOGGLE_UI_VISIBILITY` action.
https://github.com/ampproject/amphtml/compare/master...gmajoulet:hold_to_pause_system_layer?expand=1#diff-ff402d9e97126ab016345e9b77f54675R353

I took another approach than the one you suggest though, where the logic stays in this method, and the store service stays "dumb". I feel like it makes each action reusable and reliable, in case we need to hide the UI for some other feature in the future.
Happy to start thinking of how to add more logic to actions (testing strategy + where to store additional variables, like the `timer` or the `timeoutid`) if you think it's a better thing for us to do.